### DISCUSSION TITLE
[Feature] Speeding up the "naming persons" workflow

### DISCUSSION THREAD

**Discussion by Biepa** [immich-app/immich#12328] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

When doing the first install or uploading a lot of new photos and then going to add names to the faces detected I would like to propose an improvement idea to the workflow.

The current one is:
1. Click on unnamed person
2. Enter name
3. Press enter
4. Click "back" arrow
5. Check where the last named person is in the grid 
6. Click on the next one

My suggestion would be to make it possible to go to the next person once enter is pressed, or a "save and next" button (using a shortcut like "ctrl+enter") or anything similar that let me put in one name after the other with a minimal amount of steps required (without mouse interaction in the best case).

EDIT: Would be great if the "merging faces" workflow would be part of this enhancement.

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by radh21301** [comment]
Hey, I completely agree. Naming the faces and DOB is a tedious task in Immich.

For now, you can check of immich-power-tools. It is a pretty cool project, and can fasten this process by a crazy amount! 
https://github.com/varun-raj/immich-power-tools

Hopefully someday, we can have those functionality within Immich.

**Reply by Biepa** [reply]
Thank you for linking this. I'm not sure as of now though if I want to use third party tools editing my Immich data.
While the current process is a bit annoying I just started using Immich and just added the important persons for now.
Maybe I will revisit if I want it earlier in the future.

**Comment by endendino** [comment]
I wasn't able to find this while searching before (I tried!), so a mostly-overlapping post I wrote got closed. I'm pasting it here as well:

---

Like many users here, I have a large collection of photos. I have already assigned names to several hundred people using facial recognition, but I still have hundreds more to name and merge - and my unassigned faces count in the thousands.

### The Problem

The current workflow makes a methodical progression through these faces incredibly cumbersome. Often, a thumbnail is too small to tell if a face should be merged or hidden, so I need to open it to get a better look. Then, when I later navigate back, the page resets to the top and I lose my place in the grid.

### Proposed Solutions

To make this process smoother, two non-exclusive solutions come to mind (I'm sure there are others too, and I'm not sure mine are the best):
1. Save the scroll position when moving back to the main People page.
2. Add next/previous navigation so users can move directly from one person to the next without returning to the grid. Keyboard navigation could really help speed things up.

Ideally, I could open the first unassigned face and simply click "next" until I eventually finish reviewing all of them. Each unassigned face view would still have the current hide/merge options, but it would also be incredibly helpful to add a "skip"/"later review" button for people I know I should know, but need to look up their name
